### ISSUE TITLE
Mic cuts out at some point during call, every call

### ISSUE THREAD

**Issue by Peter4487** [BelledonneCommunications/linphone-android#1848]
I've been experiencing this issue for a long time with Linphone on Android

I am making a call to someone, and at some point the person can no longer hear me. I can hear the person I have called, though.

It can happen early in the call, or after 45 minutes. At some point it will happen.

I have reinstalled and tried different mic settings

Linphone Android 5.05 / Linphone SDK 5.2.16
Android 13 / Pixel 5

Logs from two instances where this happened are here (the event was towards the end of the call, of course)

1: https://privatebin.net/?a4d75dfd1cdaee99#HFvChrpmDaLEYwc7HaVZoVSDHimjVmCj69AKXbfy2fVc
2: https://privatebin.net/?e9d5616f405d0a59#vK61mXReUknyr8ypD2wDS6SMzBjMVNXGcEftzStsdJg

Password for both is lskf&uuds777sdDD

**Comment by Viish** [comment]
Hi @Peter4487,

Are the logs from the device that doesn't hear you or yours?
If they are the device that doesn't hear anything nothing explains the issue.
If they are yours then it doesn't help, I need the other device logs.

Up until the end of the call there is no apparent network issue as the audio bandwidth in both direction is > 0.
I see that you are using a VPN (that keeps notifying it's capabilities changed causing a lot of logs for nothing -_-), maybe that's what causing the issue.
We also not have a lot of experience with USB headsets on Android but there is no reason it stops working at some points, and if it did we should see it in the upload bandwidth dropping to near 0.
Issue could also come from your VoIP provider, but again there is nothing in the logs that explain the issue. Can you try with sip.linphone.org accounts?

**Comment by Peter4487** [comment]
Hi @Viish , 

1. The logs are from my device, the one running linphone. I am speaking in to the phone, but the person on the other end can't hear me. I can still hear the person on the other end, though.
2. There won't be logs from the other device, it's not running linphone. This has happened calling POTS lines, VOIP, cell phones, etc.
3. It could be the VPN. I have noticed that some mobile carriers try to block VOIP traffic...so VPN all the time :-(
4. I wasn't using a USB headset, just the default settings and the phone's integrated mic(s)
5. I could try it with sip.linphone.org, but I started the phone today and all of my linphone settings were wiped out! Perhaps because of an update?

Setting up new accounts is non-trivial, and Linphone doesn't have a way to export and import settings (naively, this seems easy to implement...). So, what I'm going to do now is try baresip for a while and see how it goes. I'm already using a different application to manage SMS since Linphone doesn't thread SMS properly (there is an issue about this, but it was closed wontfix because it's inconsistencies in message headers, IIRC...but https://github.com/michaelkourlas/voipms-sms-client does it properly somehow.

Anyway, I really appreciate your work on this, and your fast response! If I come back to it I'd be willing to do more testing.

**Comment by Viish** [comment]
> The logs are from my device, the one running linphone. I am speaking in to the phone, but the person on the other end can't hear me. I can still hear the person on the other end, though.

According to your logs the client does send audio up until the end of the call. So it might be a server issue with your VoIP provider. I suggest you contact them.

> It could be the VPN. I have noticed that some mobile carriers try to block VOIP traffic...so VPN all the time :-(

Have you tried using a WiFi network?

> I wasn't using a USB headset, just the default settings and the phone's integrated mic(s)

According to logs you do have a USB headset:
```
2023-02-07 18:38:39:507 [org.linphone/linphone-android] MESSAGE [Audio Route Helper] Playback audio currently in use is [USB-Audio - USBC Headset (AAudio) Headset]
2023-02-07 18:38:39:507 [org.linphone/linphone-android] MESSAGE [Audio Route Helper] Playback audio device currently in use is [USB-Audio - USBC Headset (AAudio) Headset]
```

> I could try it with sip.linphone.org, but I started the phone today and all of my linphone settings were wiped out! Perhaps because of an update?

No, update don't do that. It seems to be an issue on some devices (there is an open ticket for that in github), but I don't know what causes it and so far I've never been able to reproduce it...

> I'm already using a different application to manage SMS since Linphone doesn't thread SMS properly

Actually the issue is how Asterisk (and maybe other) SIP servers handle some headers.
When you don't care on separating conversations depending on accounts it's easy to handle it "properly"...

**Comment by chuckwagoncomputing** [comment]
I have the same issue, with VoIP.ms.

**Comment by Peter4487** [comment]
I am now experiencing this issue very intensely with voip.ms with the last update. Before it would happen within the first maybe 20 minutes, now it's within the first couple minutes.

The mic stops working (or at least the other pers
[linphone_logs_cleaned.txt](https://github.com/BelledonneCommunications/linphone-android/files/11533547/linphone_logs_cleaned.txt)
on can't hear me) within the first few minutes of the call. Logs attached, I sent the log shortly after the mic stopped working.

This was with linphone-android 5.0.11 (tags/5.0.11^0, release) & Linphone SDK 5.2.58 (tags/5.2.58^0, release), voip.ms, Pixel 5 running GrapheneOS.

**Comment by sk425** [comment]
I have the same issue using 
SIP Provider: voip.ms
App version: linphone 5.1.4 f-droid arm64-v8a 
OS: grapheneos 
Phone: pixel 6

However I didn't follow the troubleshooting guide here https://wiki.voip.ms/article/Call_quality_issues#One-Way_Audio

Excerpt:
-     From the account or sub account settings, select always NAT=Yes (this is the option recommended by VoIP.ms). 
-     Try using each codec in a separate way, starting with G711u codec only, from the customer portal > Main menu > Account settings > Advanced tab > allowed codecs. 
-     Only as a test, place the device in DMZ to test if the issue is related with the NAT. do not leave the device in DMZ after finishing troubleshooting. 
-     Is your router appropriate for VoIP? If you have a router and a modem, try to bypass the router to verify if the issue gets duplicated. 
-     If your router includes a SIP ALG and/or SPI Firewall setting please ensure that it is disabled. That setting is common in D-Link and Netgear routers. If this does not help make sure you are using the most recent firmware version for your device. 
-     You can see what port is being used for audio by looking at the UDP port 5060 traffic. The RTP traffic will typically be in the UDP port range 10000-20000. 


I realised two things:
1. I selected all codecs on both sides though linphone has many more than voip.ms. I then narrowed the selection on linphone to only the compatible ones. 
2. The port range was random on linphone side. I selected 42873 as the third compatible encrypted option (as an obscure option in case default sip ports are blocked on a network i'm currently on, idk lol)

Will see how things go. If I don't post back for a while, I guess we can assume it went well.
